storage/spanstore/downsampling_writer.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanstore
+
+import (
+	"hash"
+	"hash/fnv"
+	"math"
+	"sync"
+
+	"github.com/uber/jaeger-lib/metrics"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+const (
+	traceIDByteSize = 16
```

Comment:
shouldn't hardcode this here. The `model/ids.go` file has this:
```
// Size returns the size of this datum in protobuf. It is always 16 bytes.
func (t *TraceID) Size() int {
        return 16
}
```

You can create a fake trace id and ask it for Size(): 
```
var (
    traceIDByteSize := model.NewTraceID(0, 0).Size()`
)
```